core/src/main/java/io/grpc/internal/CensusStatsModule.java

```diff
@@ -154,18 +155,58 @@ ClientInterceptor getClientInterceptor(boolean recordStartedRpcs, boolean record
-        AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "outboundWireSize");
-    private static final AtomicLongFieldUpdater<ClientTracer> inboundWireSizeUpdater =
-        AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "inboundWireSize");
-    private static final AtomicLongFieldUpdater<ClientTracer> outboundUncompressedSizeUpdater =
-        AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "outboundUncompressedSize");
-    private static final AtomicLongFieldUpdater<ClientTracer> inboundUncompressedSizeUpdater =
-        AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "inboundUncompressedSize");
+    @Nullable private static final AtomicLongFieldUpdater<ClientTracer> outboundMessageCountUpdater;
+    @Nullable private static final AtomicLongFieldUpdater<ClientTracer> inboundMessageCountUpdater;
+    @Nullable private static final AtomicLongFieldUpdater<ClientTracer> outboundWireSizeUpdater;
+    @Nullable private static final AtomicLongFieldUpdater<ClientTracer> inboundWireSizeUpdater;
+
+    @Nullable
+    private static final AtomicLongFieldUpdater<ClientTracer> outboundUncompressedSizeUpdater;
+
+    @Nullable
+    private static final AtomicLongFieldUpdater<ClientTracer> inboundUncompressedSizeUpdater;
+
+    /**
+     * When using Atomic*FieldUpdater, some Samsung Android 5.0.x devices encounter a bug in their
+     * JDK reflection API that triggers a NoSuchFieldException. When this occurs, we fallback to
+     * (potentially racy) direct updates of the volatile variables.
+     */
+    static {
+      AtomicLongFieldUpdater<ClientTracer> tmpOutboundMessageCountUpdater;
+      AtomicLongFieldUpdater<ClientTracer> tmpInboundMessageCountUpdater;
+      AtomicLongFieldUpdater<ClientTracer> tmpOutboundWireSizeUpdater;
+      AtomicLongFieldUpdater<ClientTracer> tmpInboundWireSizeUpdater;
+      AtomicLongFieldUpdater<ClientTracer> tmpOutboundUncompressedSizeUpdater;
+      AtomicLongFieldUpdater<ClientTracer> tmpInboundUncompressedSizeUpdater;
+      try {
+        tmpOutboundMessageCountUpdater =
+            AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "outboundMessageCount");
+        tmpInboundMessageCountUpdater =
+            AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "inboundMessageCount");
+        tmpOutboundWireSizeUpdater =
+            AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "outboundWireSize");
+        tmpInboundWireSizeUpdater =
+            AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "inboundWireSize");
+        tmpOutboundUncompressedSizeUpdater =
+            AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "outboundUncompressedSize");
+        tmpInboundUncompressedSizeUpdater =
+            AtomicLongFieldUpdater.newUpdater(ClientTracer.class, "inboundUncompressedSize");
+      } catch (Throwable t) {
+        logger.log(Level.WARNING, "Creating atomic field updaters failed", t);
+        tmpOutboundMessageCountUpdater = null;
```

Comment:
initialize all these to null above?

Reply:
Leaving as-is so that all field updater references are assigned a value or they are all null